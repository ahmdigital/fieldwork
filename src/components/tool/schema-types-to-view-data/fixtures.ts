const fixtures = [
  {
    fields: [
      { description: 'this name is different for some reason', name: 'name', type: 'string' },
      {
        name: 'slogan',
        of: [
          {
            lists: [],
            marks: { annotations: [], decorators: [{ title: 'Emphasis', value: 'em' }] },
            styles: [{ title: 'Normal', value: 'normal' }],
            type: 'block',
          },
        ],
        type: 'array',
      },
      { name: 'description', of: [{ type: 'block' }], type: 'array' },
    ],
    name: 'brandCampaign',
    type: 'document',
  },
  {
    fields: [
      { name: 'name', type: 'string' },
      { name: 'description', of: [{ type: 'block' }], type: 'array' },
    ],
    name: 'legislation',
    type: 'document',
  },
  {
    fields: [
      { name: 'name', type: 'string' },
      { name: 'identifier', options: { source: 'name' }, type: 'slug' },
      { name: 'legalName', type: 'string' },
      { name: 'logo', type: 'image' },
    ],
    name: 'organisation',
    type: 'document',
  },
  {
    fields: [
      { name: 'name', type: 'string' },
      { name: 'identifier', options: { source: 'name' }, type: 'slug' },
    ],
    name: 'placeholder',
    type: 'document',
  },
  {
    fields: [
      { name: 'name', type: 'string' },
      { name: 'startTime', options: { dateFormat: 'D MMM YYYY', timeStep: 1 }, type: 'datetime' },
      { name: 'endTime', options: { dateFormat: 'D MMM YYYY', timeStep: 1 }, type: 'datetime' },
    ],
    name: 'release',
    preview: { select: { endTime: 'endTime', startTime: 'startTime', title: 'name' } },
    type: 'document',
  },
  {
    fields: [
      { name: 'name', type: 'string' },
      { name: 'identifier', options: { source: 'name' }, type: 'slug' },
      {
        name: 'slogan',
        of: [
          {
            lists: [],
            marks: { annotations: [], decorators: [{ title: 'Emphasis', value: 'em' }] },
            styles: [{ title: 'Normal', value: 'normal' }],
            type: 'block',
          },
        ],
        type: 'array',
      },
      { name: 'description', of: [{ type: 'block' }], type: 'array' },
    ],
    name: 'service',
    type: 'document',
  },
  {
    fields: [
      { name: 'name', type: 'string' },
      {
        name: 'slogan',
        of: [
          {
            lists: [],
            marks: { annotations: [], decorators: [{ title: 'Emphasis', value: 'em' }] },
            styles: [{ title: 'Normal', value: 'normal' }],
            type: 'block',
          },
        ],
        type: 'array',
      },
      { name: 'description', of: [{ type: 'block' }], type: 'array' },
      { name: 'providesServices', of: [{ to: [{ type: 'service' }], type: 'reference' }], type: 'array' },
    ],
    name: 'serviceChannel',
    type: 'document',
  },
];

export default fixtures;
