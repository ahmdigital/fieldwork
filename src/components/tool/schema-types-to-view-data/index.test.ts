import fixtures from './fixtures';
import schemaTypesToViewData from '.';

it('maps schema types array to view data', () => {
  expect(schemaTypesToViewData(fixtures)).toMatchSnapshot();
});
