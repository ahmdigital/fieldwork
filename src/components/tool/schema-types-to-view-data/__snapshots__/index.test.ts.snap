// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maps schema types array to view data 1`] = `
{
  "description": {
    "63188285": [
      {
        "name": "description",
        "of": [
          {
            "type": "block",
          },
        ],
        "type": "array",
        "typeName": "serviceChannel",
        "validation": [],
      },
      {
        "name": "description",
        "of": [
          {
            "type": "block",
          },
        ],
        "type": "array",
        "typeName": "service",
        "validation": [],
      },
      {
        "name": "description",
        "of": [
          {
            "type": "block",
          },
        ],
        "type": "array",
        "typeName": "legislation",
        "validation": [],
      },
      {
        "name": "description",
        "of": [
          {
            "type": "block",
          },
        ],
        "type": "array",
        "typeName": "brandCampaign",
        "validation": [],
      },
    ],
  },
  "endTime": {
    "397541495": [
      {
        "name": "endTime",
        "options": {
          "dateFormat": "D MMM YYYY",
          "timeStep": 1,
        },
        "type": "datetime",
        "typeName": "release",
        "validation": [],
      },
    ],
  },
  "identifier": {
    "-1209135794": [
      {
        "name": "identifier",
        "options": {
          "source": "name",
        },
        "type": "slug",
        "typeName": "service",
        "validation": [],
      },
      {
        "name": "identifier",
        "options": {
          "source": "name",
        },
        "type": "slug",
        "typeName": "placeholder",
        "validation": [],
      },
      {
        "name": "identifier",
        "options": {
          "source": "name",
        },
        "type": "slug",
        "typeName": "organisation",
        "validation": [],
      },
    ],
  },
  "legalName": {
    "-68257741": [
      {
        "name": "legalName",
        "type": "string",
        "typeName": "organisation",
        "validation": [],
      },
    ],
  },
  "logo": {
    "-318825570": [
      {
        "name": "logo",
        "type": "image",
        "typeName": "organisation",
        "validation": [],
      },
    ],
  },
  "name": {
    "-1080650659": [
      {
        "description": "this name is different for some reason",
        "name": "name",
        "type": "string",
        "typeName": "brandCampaign",
        "validation": [],
      },
    ],
    "317756900": [
      {
        "name": "name",
        "type": "string",
        "typeName": "serviceChannel",
        "validation": [],
      },
      {
        "name": "name",
        "type": "string",
        "typeName": "service",
        "validation": [],
      },
      {
        "name": "name",
        "type": "string",
        "typeName": "release",
        "validation": [],
      },
      {
        "name": "name",
        "type": "string",
        "typeName": "placeholder",
        "validation": [],
      },
      {
        "name": "name",
        "type": "string",
        "typeName": "organisation",
        "validation": [],
      },
      {
        "name": "name",
        "type": "string",
        "typeName": "legislation",
        "validation": [],
      },
    ],
  },
  "providesServices": {
    "-226513091": [
      {
        "name": "providesServices",
        "of": [
          {
            "to": [
              {
                "type": "service",
              },
            ],
            "type": "reference",
          },
        ],
        "type": "array",
        "typeName": "serviceChannel",
        "validation": [],
      },
    ],
  },
  "slogan": {
    "-1681617650": [
      {
        "name": "slogan",
        "of": [
          {
            "lists": [],
            "marks": {
              "annotations": [],
              "decorators": [
                {
                  "title": "Emphasis",
                  "value": "em",
                },
              ],
            },
            "styles": [
              {
                "title": "Normal",
                "value": "normal",
              },
            ],
            "type": "block",
          },
        ],
        "type": "array",
        "typeName": "serviceChannel",
        "validation": [],
      },
      {
        "name": "slogan",
        "of": [
          {
            "lists": [],
            "marks": {
              "annotations": [],
              "decorators": [
                {
                  "title": "Emphasis",
                  "value": "em",
                },
              ],
            },
            "styles": [
              {
                "title": "Normal",
                "value": "normal",
              },
            ],
            "type": "block",
          },
        ],
        "type": "array",
        "typeName": "service",
        "validation": [],
      },
      {
        "name": "slogan",
        "of": [
          {
            "lists": [],
            "marks": {
              "annotations": [],
              "decorators": [
                {
                  "title": "Emphasis",
                  "value": "em",
                },
              ],
            },
            "styles": [
              {
                "title": "Normal",
                "value": "normal",
              },
            ],
            "type": "block",
          },
        ],
        "type": "array",
        "typeName": "brandCampaign",
        "validation": [],
      },
    ],
  },
  "startTime": {
    "-501757616": [
      {
        "name": "startTime",
        "options": {
          "dateFormat": "D MMM YYYY",
          "timeStep": 1,
        },
        "type": "datetime",
        "typeName": "release",
        "validation": [],
      },
    ],
  },
}
`;
